In [2]:
import pandas as pd

In [4]:
data=pd.read_csv('digitalanalytics.csv')

In [5]:
data

id        time           created_at   from_user_name  \
0    828531683381100544  1486372321  2017-02-06 09:12:01      AT_Internet   
1    828536966031380481  1486373580  2017-02-06 09:33:00   AT_Internet_DE   
2    828540415360004096  1486374403  2017-02-06 09:46:43         ajmuguia   
3    828540731975401473  1486374478  2017-02-06 09:47:58         ajmuguia   
4    828540960443351041  1486374533  2017-02-06 09:48:53         ajmuguia   
5    828542296274661376  1486374851  2017-02-06 09:54:11    pascalrossini   
6    828544758989602816  1486375438  2017-02-06 10:03:58  MarketingJobs4U   
7    828553689140981760  1486377567  2017-02-06 10:39:27         guzogang   
8    828556421822312449  1486378219  2017-02-06 10:50:19     ShikhaBakshi   
9    828559249773060096  1486378893  2017-02-06 11:01:33       DIMHouston   
10   828565053674749952  1486380277  2017-02-06 11:24:37    Fernanrelenia   
11   828565322907131904  1486380341  2017-02-06 11:25:41    Fernanrelenia   
12   828565538926362624  1486380393  2017-02-06 11:26:33    Fernanrelenia   
13   828576889560109057  1486383099  2017-02-06 12:11:39         Optizent   
14   828587896756174848  1486385723  2017-02-06 12:55:23     GonzaloGarde   
15   828588105867333632  1486385773  2017-02-06 12:56:13     GonzaloGarde   
16   828592894630096897  1486386915  2017-02-06 13:15:15      Jobs_Koeln_   
17   828619425419030529  1486393240  2017-02-06 15:00:40  HeadHunterJames   
18   828623455239036929  1486394201  2017-02-06 15:16:41      tmj_aus_adv   
19   828644192503132161  1486399145  2017-02-06 16:39:05  FRreferencement   
20   828645687625707520  1486399502  2017-02-06 16:45:02   LesCommuniques   
21   828645723738619905  1486399510  2017-02-06 16:45:10    capes_tunisie   
22   828655123018633216  1486401751  2017-02-06 17:22:31         Optizent   
23   828655600334729216  1486401865  2017-02-06 17:24:25         main_roi   
24   828660295165997056  1486402984  2017-02-06 17:43:04     nicolemleite   
25   828664538157424641  1486403996  2017-02-06 17:59:56   TopTalentMatch   
26   828670925893664769  1486405519  2017-02-06 18:25:19       HoloMarkeD   
27   828671243708665857  1486405595  2017-02-06 18:26:35          estortz   
28   828674948960972800  1486406478  2017-02-06 18:41:18   EricP_Officiel   
29   828678402936803329  1486407302  2017-02-06 18:55:02       redcatandy   
..                  ...         ...                  ...              ...   
411  831449532244205569  1487067990  2017-02-14 10:26:30   herbtheentspec   
412  831450973117743104  1487068334  2017-02-14 10:32:14      Stingraytwo   
413  831454062558208000  1487069070  2017-02-14 10:44:30  rajchatting0009   
414  831454251159257088  1487069115  2017-02-14 10:45:15           SJS_US   
415  831455673632296962  1487069454  2017-02-14 10:50:54         Datafloq   
416  831455690447286272  1487069458  2017-02-14 10:50:58          EAGCOPS   
417  831456209689509888  1487069582  2017-02-14 10:53:02      VanRijmenam   
418  831460982866247681  1487070720  2017-02-14 11:12:00         desnicky   
419  831464316104822784  1487071515  2017-02-14 11:25:15           SJS_US   
420  831465534734626816  1487071806  2017-02-14 11:30:06           SJS_US   
421  831468031528288264  1487072401  2017-02-14 11:40:01   itsax_praktika   
422  831480708610846720  1487075423  2017-02-14 12:30:23  ReallyReally_Me   
423  831481515641016320  1487075616  2017-02-14 12:33:36     Designerdock   
424  831956876845842434  1487188951  2017-02-15 20:02:31      kstansberry   
425  831972424484675586  1487192657  2017-02-15 21:04:17         Optizent   
426  831974869709512709  1487193240  2017-02-15 21:14:00      MamzelleMNa   
427  831982316461363202  1487195016  2017-02-15 21:43:36     nicolemleite   
428  832000216719097857  1487199284  2017-02-15 22:54:44    AnalyticsWebs   
429  832019329709273088  1487203841  2017-02-16 00:10:41         Optizent   
430  832019800943640576  1487203953  2017-02-16 00:12:33       simple8job   
431  832021024963

1. **Basic descriptives**

In [4]:
data.describe()

id          time  possibly_sensitive  withheld_copyright  \
count  4.410000e+02  4.410000e+02          409.000000                 0.0   
mean   8.299928e+17  1.486721e+09            0.012225                 NaN   
std    9.688971e+14  2.310031e+05            0.110023                 NaN   
min    8.285317e+17  1.486372e+09            0.000000                 NaN   
25%    8.291964e+17  1.486531e+09            0.000000                 NaN   
50%    8.297821e+17  1.486670e+09            0.000000                 NaN   
75%    8.309157e+17  1.486941e+09            0.000000                 NaN   
max    8.321202e+17  1.487228e+09            1.000000                 NaN   

       withheld_scope  truncated  retweet_count  favorite_count  \
count             0.0      441.0          441.0           441.0   
mean              NaN        0.0            0.0             0.0   
std               NaN        0.0            0.0             0.0   
min               NaN        0.0            0.0             0.0   
25%               NaN        0.0            0.0             0.0   
50%               NaN        0.0            0.0             0.0   
75%               NaN        0.0            0.0             0.0   
max               NaN        0.0            0.0             0.0   

       in_reply_to_status_id  quoted_status_id            ...             \
count                    0.0               0.0            ...              
mean                     NaN               NaN            ...              
std                      NaN               NaN            ...              
min                      NaN               NaN            ...              
25%                      NaN               NaN            ...              
50%                      NaN               NaN            ...              
75%                      NaN               NaN            ...              
max                      NaN               NaN            ...              

              lng  from_user_id  from_user_verified  from_user_utcoffset  \
count    8.000000  4.410000e+02          441.000000           344.000000   
mean    33.280653  8.456917e+16            0.015873        -11569.186047   
std     97.635795  2.425224e+17            0.125126         19229.104624   
min    -97.743057  9.947610e+05            0.000000        -36000.000000   
25%    -75.840582  2.941688e+07            0.000000        -28800.000000   
50%    103.867744  4.018025e+08            0.000000        -18000.000000   
75%    103.867744  2.670613e+09            0.000000          3600.000000   
max    103.867744  8.300623e+17            1.000000         39600.000000   

       from_user_tweetcount  from_user_followercount  from_user_friendcount  \
count          4.410000e+02               441.000000             441.000000   
mean           5.470849e+04              2883.167800            1755.482993   
std            1.614955e+05              9178.667744            4063.098513   
min            3.000000e+00                 0.000000               0.000000   
25%            2.896000e+03               445.000000             197.000000   
50%            1.334800e+04               993.000000            1105.000000   
75%            3.425100e+04              2126.000000            1402.000000   
max            1.777822e+06            150934.000000           56658.000000   

       from_user_favourites_count  from_user_listed  from_user_withheld_scope  
count                  441.000000        441.000000                       0.0  
mean                  8715.498866        616.083900                       NaN  
std                  33973.951728       1146.656196                       NaN  
min                      0.000000          0.000000                       NaN  
25%                      6.000000         95.000000                       NaN  
50%                     37.000000        146.000000                       NaN  
75%                    490.000000        746.000000                       NaN

Description:

*How many tweets are there?* **441**

*What is the average number of friends and of followers that users tweeting about your topic have?* **friends mean = 1755.5; followers mean = 2883.2**

*What is the average number of retweets that each tweet got?* **0**


I think there should be a normal, simple way to call the needed description for whatever we need, how ever when I tried to do it (as below to answer the 2nd question), the result didn't seem ok. 

*What is the average number of friends and of followers that users tweeting about your topic have?*

In [17]:
data.groupby('from_user_friendcount').describe()

favorite_count                                     \
                               count mean  std  min  25%  50%  75%  max   
from_user_friendcount                                                     
0                               11.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1                                4.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
2                                2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
3                                5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
4                                4.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
6                                5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
7                                2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
8                                1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
9                                6.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
15                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
16                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
18                               4.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
21                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
22                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
23                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
25                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
26                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
28                               4.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
29                               2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
31                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
34                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
36                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
38                               2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
39                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
40                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
41                               3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
42                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
43                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
44                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
55                               1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
...                              ...  ...  ...  ...  ...  ...  ...  ...   
4887                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
4897                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
4952                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
4958                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
4959                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
4996                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5000                             2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
5001                             2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
5050                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5436                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5476                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5539                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5595                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5666                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5751                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
5946                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
6386                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
7054                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
7702                             1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
9182                             2.0  0.0  0.0 

*How many tweets are there per language?*

In [10]:
data.groupby('from_user_lang').describe()

favorite_count                                     \
                        count mean  std  min  25%  50%  75%  max   
from_user_lang                                                     
de                       23.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
en                      374.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
en-gb                     3.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
es                       17.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
fr                       16.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
it                        1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
ja                        1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
nl                        1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
pt                        4.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ro                        1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   

               from_user_favourites_count               ...  \
                                    count          mean ...   
from_user_lang                                          ...   
de                                   23.0     85.173913 ...   
en                                  374.0   6539.066845 ...   
en-gb                                 3.0    388.666667 ...   
es                                   17.0  74545.941176 ...   
fr                                   16.0   4327.875000 ...   
it                                    1.0    250.000000 ...   
ja                                    1.0      0.000000 ...   
nl                                    1.0  30888.000000 ...   
pt                                    4.0   6781.750000 ...   
ro                                    1.0      7.000000 ...   

               withheld_copyright     withheld_scope                           \
                              75% max          count mean std min 25% 50% 75%   
from_user_lang                                                                  
de                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
en                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
en-gb                         NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
es                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
fr                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
it                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
ja                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
nl                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
pt                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   
ro                            NaN NaN            0.0  NaN NaN NaN NaN NaN NaN   

                    
               max  
from_user_lang      
de             NaN  
en             NaN  
en-gb          NaN  
es             NaN  
fr             NaN  
it             NaN  
ja             NaN  
nl             NaN  
pt             NaN  
ro             NaN  

[10 rows x 168 columns]

 **Sentiment analysis**

In [6]:
pd.DataFrame(data)

id        time           created_at   from_user_name  \
0    828531683381100544  1486372321  2017-02-06 09:12:01      AT_Internet   
1    828536966031380481  1486373580  2017-02-06 09:33:00   AT_Internet_DE   
2    828540415360004096  1486374403  2017-02-06 09:46:43         ajmuguia   
3    828540731975401473  1486374478  2017-02-06 09:47:58         ajmuguia   
4    828540960443351041  1486374533  2017-02-06 09:48:53         ajmuguia   
5    828542296274661376  1486374851  2017-02-06 09:54:11    pascalrossini   
6    828544758989602816  1486375438  2017-02-06 10:03:58  MarketingJobs4U   
7    828553689140981760  1486377567  2017-02-06 10:39:27         guzogang   
8    828556421822312449  1486378219  2017-02-06 10:50:19     ShikhaBakshi   
9    828559249773060096  1486378893  2017-02-06 11:01:33       DIMHouston   
10   828565053674749952  1486380277  2017-02-06 11:24:37    Fernanrelenia   
11   828565322907131904  1486380341  2017-02-06 11:25:41    Fernanrelenia   
12   828565538926362624  1486380393  2017-02-06 11:26:33    Fernanrelenia   
13   828576889560109057  1486383099  2017-02-06 12:11:39         Optizent   
14   828587896756174848  1486385723  2017-02-06 12:55:23     GonzaloGarde   
15   828588105867333632  1486385773  2017-02-06 12:56:13     GonzaloGarde   
16   828592894630096897  1486386915  2017-02-06 13:15:15      Jobs_Koeln_   
17   828619425419030529  1486393240  2017-02-06 15:00:40  HeadHunterJames   
18   828623455239036929  1486394201  2017-02-06 15:16:41      tmj_aus_adv   
19   828644192503132161  1486399145  2017-02-06 16:39:05  FRreferencement   
20   828645687625707520  1486399502  2017-02-06 16:45:02   LesCommuniques   
21   828645723738619905  1486399510  2017-02-06 16:45:10    capes_tunisie   
22   828655123018633216  1486401751  2017-02-06 17:22:31         Optizent   
23   828655600334729216  1486401865  2017-02-06 17:24:25         main_roi   
24   828660295165997056  1486402984  2017-02-06 17:43:04     nicolemleite   
25   828664538157424641  1486403996  2017-02-06 17:59:56   TopTalentMatch   
26   828670925893664769  1486405519  2017-02-06 18:25:19       HoloMarkeD   
27   828671243708665857  1486405595  2017-02-06 18:26:35          estortz   
28   828674948960972800  1486406478  2017-02-06 18:41:18   EricP_Officiel   
29   828678402936803329  1486407302  2017-02-06 18:55:02       redcatandy   
..                  ...         ...                  ...              ...   
411  831449532244205569  1487067990  2017-02-14 10:26:30   herbtheentspec   
412  831450973117743104  1487068334  2017-02-14 10:32:14      Stingraytwo   
413  831454062558208000  1487069070  2017-02-14 10:44:30  rajchatting0009   
414  831454251159257088  1487069115  2017-02-14 10:45:15           SJS_US   
415  831455673632296962  1487069454  2017-02-14 10:50:54         Datafloq   
416  831455690447286272  1487069458  2017-02-14 10:50:58          EAGCOPS   
417  831456209689509888  1487069582  2017-02-14 10:53:02      VanRijmenam   
418  831460982866247681  1487070720  2017-02-14 11:12:00         desnicky   
419  831464316104822784  1487071515  2017-02-14 11:25:15           SJS_US   
420  831465534734626816  1487071806  2017-02-14 11:30:06           SJS_US   
421  831468031528288264  1487072401  2017-02-14 11:40:01   itsax_praktika   
422  831480708610846720  1487075423  2017-02-14 12:30:23  ReallyReally_Me   
423  831481515641016320  1487075616  2017-02-14 12:33:36     Designerdock   
424  831956876845842434  1487188951  2017-02-15 20:02:31      kstansberry   
425  831972424484675586  1487192657  2017-02-15 21:04:17         Optizent   
426  831974869709512709  1487193240  2017-02-15 21:14:00      MamzelleMNa   
427  831982316461363202  1487195016  2017-02-15 21:43:36     nicolemleite   
428  832000216719097857  1487199284  2017-02-15 22:54:44    AnalyticsWebs   
429  832019329709273088  1487203841  2017-02-16 00:10:41         Optizent   
430  832019800943640576  1487203953  2017-02-16 00:12:33       simple8job   
431  832021024963

In [7]:
df_data=pd.DataFrame(data)

In [27]:
df_data.columns

Index(['id', 'time', 'created_at', 'from_user_name', 'text', 'filter_level',
       'possibly_sensitive', 'withheld_copyright', 'withheld_scope',
       'truncated', 'retweet_count', 'favorite_count', 'lang', 'to_user_name',
       'in_reply_to_status_id', 'quoted_status_id', 'source', 'location',
       'lat', 'lng', 'from_user_id', 'from_user_realname',
       'from_user_verified', 'from_user_description', 'from_user_url',
       'from_user_profile_image_url', 'from_user_utcoffset',
       'from_user_timezone', 'from_user_lang', 'from_user_tweetcount',
       'from_user_followercount', 'from_user_friendcount',
       'from_user_favourites_count', 'from_user_listed',
       'from_user_withheld_scope', 'from_user_created_at'],
      dtype='object')

In [11]:
df_filtered=df_data[df_data['from_user_lang']=='en']

In [13]:
df_filtered.describe()

id          time  possibly_sensitive  withheld_copyright  \
count  3.740000e+02  3.740000e+02          347.000000                 0.0   
mean   8.299913e+17  1.486720e+09            0.008646                 NaN   
std    9.531119e+14  2.272396e+05            0.092712                 NaN   
min    8.285404e+17  1.486374e+09            0.000000                 NaN   
25%    8.291933e+17  1.486530e+09            0.000000                 NaN   
50%    8.298195e+17  1.486679e+09            0.000000                 NaN   
75%    8.308925e+17  1.486935e+09            0.000000                 NaN   
max    8.321202e+17  1.487228e+09            1.000000                 NaN   

       withheld_scope  truncated  retweet_count  favorite_count  \
count             0.0      374.0          374.0           374.0   
mean              NaN        0.0            0.0             0.0   
std               NaN        0.0            0.0             0.0   
min               NaN        0.0            0.0             0.0   
25%               NaN        0.0            0.0             0.0   
50%               NaN        0.0            0.0             0.0   
75%               NaN        0.0            0.0             0.0   
max               NaN        0.0            0.0             0.0   

       in_reply_to_status_id  quoted_status_id            ...             \
count                    0.0               0.0            ...              
mean                     NaN               NaN            ...              
std                      NaN               NaN            ...              
min                      NaN               NaN            ...              
25%                      NaN               NaN            ...              
50%                      NaN               NaN            ...              
75%                      NaN               NaN            ...              
max                      NaN               NaN            ...              

              lng  from_user_id  from_user_verified  from_user_utcoffset  \
count    8.000000  3.740000e+02          374.000000           300.000000   
mean    33.280653  7.495103e+16            0.018717        -12606.000000   
std     97.635795  2.301594e+17            0.135704         19763.297391   
min    -97.743057  7.491782e+06            0.000000        -36000.000000   
25%    -75.840582  2.413248e+07            0.000000        -28800.000000   
50%    103.867744  1.251947e+08            0.000000        -21600.000000   
75%    103.867744  2.479525e+09            0.000000             0.000000   
max    103.867744  8.288964e+17            1.000000         39600.000000   

       from_user_tweetcount  from_user_followercount  from_user_friendcount  \
count          3.740000e+02               374.000000             374.000000   
mean           4.419566e+04              2991.644385            1803.058824   
std            1.471662e+05              9759.024845            4163.460982   
min            3.000000e+00                 1.000000               0.000000   
25%            2.094250e+03               473.000000             227.250000   
50%            1.267600e+04               993.000000            1323.000000   
75%            2.483900e+04              2125.000000            1405.750000   
max            1.777822e+06            150934.000000           56658.000000   

       from_user_favourites_count  from_user_listed  from_user_withheld_scope  
count                  374.000000        374.000000                       0.0  
mean                  6539.066845        592.868984                       NaN  
std                  29453.997459       1104.858875                       NaN  
min                      0.000000          0.000000                       NaN  
25%                      6.000000         97.000000                       NaN  
50%                     34.500000        133.000000                       NaN  
75%                    482.500000        759.000000                       NaN

In [28]:
df_filtered[['id','text','from_user_lang']]

id                                               text  \
2    828540415360004096  RT @Loui_Picard: Manage up with Digital Analyt...   
3    828540731975401473  RT @jose_garde: Manage up with Digital Analyti...   
4    828540960443351041  RT @JoanIratxeta: Manage up with Digital Analy...   
6    828544758989602816  Senior Digital Analytics Manager - £70,000 (+1...   
7    828553689140981760  Senior Digital Analytics Manager - £70,000 (+1...   
8    828556421822312449  Fractal Analytics is hiring a Principal Consul...   
9    828559249773060096  Test Your Mettle: Adobe Digital Analytics Comp...   
10   828565053674749952  RT @Loui_Picard: Manage up with Digital Analyt...   
11   828565322907131904  RT @jose_garde: Manage up with Digital Analyti...   
12   828565538926362624  RT @JoanIratxeta: Manage up with Digital Analy...   
13   828576889560109057  Digital Analytics Speakers Required https://t....   
17   828619425419030529  Senior Analyst (Digital Analytics) - Rogers Co...   
18   828623455239036929  Can you recommend anyone for this #job? Senior...   
22   828655123018633216  Senior Digital Analytics Specialist at Consume...   
24   828660295165997056  Digital Analytics Weekly Update is out! https:...   
25   828664538157424641  Wanted: Digital Analytics talents to bring in ...   
26   828670925893664769  @_robindickinson I think the argument is digit...   
27   828671243708665857  RT @DeVryCareers: Do you love data and digital...   
29   828678402936803329  Head of Customer/Digital Analytics #job in #Lo...   
30   828685452223787008  Explore A Career in Digital Analytics https://...   
32   828699952360468480  #measure Senior Analyst, Digital Analytics Job...   
33   828704074983960578  Digital analytics industry veteran roundup: #w...   
34   828720830918033408  Digital Analytics Weekly Update is out! https:...   
35   828732615284891648  5 Marketing blind spots that Digital Analytics...   
36   828734517179359232  Digital Analytics Manager in San Francisco, CA...   
37   828739103135059972  RT @garyangel: I'm doing a last expert-level d...   
38   828742906643103745  Chart form follows data. Don't choose a format...   
39   828749693404405760  #measure Manager, Digital Analytics at MEC, Ne...   
40   828777463400583172  SEO - Compare 16 leading digital analytics pla...   
41   828793206263336962  Can you recommend anyone for this #job? Digita...   
..                  ...                                                ...   
402  831406235207471104  Digital Analytics Association Symposium i http...   
403  831406309589340160  Digital Analytics Association Symposium i http...   
406  831429796169183233  RT @LEGO_Careers: See our latest #Singapore #j...   
408  831443383239528451  Manager Digital Analytics Implementation @DowJ...   
409  831445985909837824  See our latest #Singapore #job and click to ap...   
410  831448818243739648  Got a burning ❤️ for analytics? Join the Digit...   
411  831449532244205569  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
412  831450973117743104  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
413  831454062558208000  Circus Social partners with e27 for digital an...   
414  831454251159257088  Manager, Digital Analytics Implementation, #Es...   
416  831455690447286272  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
417  831456209689509888  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
418  831460982866247681  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
419  831464316104822784  Manager, Streaming and Digital Analytics, #Bur...   
420  831465534734626816  Manager, Streaming and Digital Analytics, #San...   
422  831480708610846720  RT @BCCare: Got a burning ❤️ for analytics? Jo...   
424  831956876845842434  Here for today's #prprofs chat on #fakenews. P...   
425  831972424484675586  #measure Manager, Digital Analytics at MEC, Ne...   
427  831982316461363202  Digital Analytics Weekly Update is out! https:...   
429  832019329709273088  Digital Analytics 

In [23]:
data_pkl=df_filtered.filter(items=['id','text'])

In [25]:
data_pkl.to_pickle('NhungPham_EN.pkl')

**Trial sentiment analysis**

I misread the challenge so I did some search on how to do sentiment analysis. Below is what I tried though they didnt generate right results.

In [31]:
import re

In [40]:
def tweet_sentiment():
    analysis=df_data('text')
    if analysis.sentiment.popularity>0:
        return 'positive'
    elif analysis.sentiment.popularity==0:
        return 'neutral'
    else:
        return 'negative'

In [48]:
tweet=df_data.groupby('text')

In [49]:
ptweets=[tweet for tweet in tweet if tweet_sentiment=='positive']
print('Percentage of positive tweets:', 100*len(ptweets)/len(tweet),'%')

Percentage of positive tweets: 0.0 %


In [50]:
ntweets=[tweet for tweet in tweet if tweet_sentiment=='negative']
print('Percentage of negative tweets:', 100*len(ntweets)/len(tweet),'%')

Percentage of negative tweets: 0.0 %


In [51]:
neutweets=[tweet for tweet in tweet if tweet_sentiment=='neutral']
print('Percentage of neutral tweets:', 100*len(neutweets)/len(tweet),'%')

Percentage of neutral tweets: 0.0 %
